cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -1544,11 +1655,11 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 cond
             };
 
-            ctx.emit(Inst::TrapIf {
+            ctx.emit_safepoint(Inst::TrapIf {
+                // FIXME JRS: emit or emit_safepoint?
```

Comment:
They are non-resumable; no stackmap needed.  Choosing plain `emit` therefore.

Reply:
.. but then, why does `Opcode::Trap` use `emit_safepoint` ?

Reply:
One thing that concerns me here is that it seems like we're conflating the semantics of wasm with the semantics of CLIF.  Whether or not `Opcode::Trapif | Opcode::Trapff` produces a safepoint should surely be specified as part of CLIF's semantics, and not be dependent of the semantics of whatever that CLIF was translated from.

Reply:
Hmm, I think I missed some context previously -- I thought for some reason that your additions here only had to do with atomics-related misalignments, but of course if I scroll up just a bit I see this is for any `Opcode::Trap{if,ff}`. D'oh. Yes, it should always have a safepoint if it is a trap at the CLIF level; the only non-safepoint-adorned traps are the ones that are internal to e.g. divides. I suppose we should specify in the CLIF descriptions that any traps generated by instruction error conditions (rather than explicit trap instructions) will not have safepoints.

Just to be sure -- @bnjbvr, could you confirm that we don't expect (or can get away with specifying) that there is no safepoint data for e.g. divide traps?

Reply:
Changing to `emit_safepoint`.

Reply:
> Just to be sure -- @bnjbvr, could you confirm that we don't expect (or can get away with specifying) that there is no safepoint data for e.g. divide traps?

Agreed, and i think that indeed it should be specified that no GC must happen during stack unwinding (that sounds very scary and unsafe anyways)(now I wonder if wasm exception handling + wasm GC types could make this case actually possible).